src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -11761,63 +11813,57 @@ private bool CanStartConsequenceExpression()
-                SyntaxKind.OpenBracketToken => _syntaxFactory.ElementBindingExpression(this.ParseBracketedArgumentList()),
-                _ => throw ExceptionUtilities.Unreachable(),
-            };
-
-            while (true)
+            ExpressionSyntax parseWhenNotNull(ExpressionSyntax expr)
             {
-                // Nullable suppression operators should only be consumed by a conditional access
-                // if there are further conditional operations performed after the suppression
-                if (isOptionalExclamationsFollowedByConditionalOperation())
-                {
-                    while (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
-                        expr = _syntaxFactory.PostfixUnaryExpression(SyntaxKind.SuppressNullableWarningExpression, expr, EatToken());
-                }
-
-                switch (this.CurrentToken.Kind)
+                while (true)
                 {
-                    case SyntaxKind.OpenParenToken:
-                        expr = _syntaxFactory.InvocationExpression(expr, this.ParseParenthesizedArgumentList());
-                        continue;
-
-                    case SyntaxKind.OpenBracketToken:
-                        expr = _syntaxFactory.ElementAccessExpression(expr, this.ParseBracketedArgumentList());
-                        continue;
+                    // Nullable suppression operators should only be consumed by a conditional access
+                    // if there are further conditional operations performed after the suppression
+                    if (isOptionalExclamationsFollowedByConditionalOperation())
+                    {
+                        while (this.CurrentToken.Kind == SyntaxKind.ExclamationToken)
+                            expr = _syntaxFactory.PostfixUnaryExpression(SyntaxKind.SuppressNullableWarningExpression, expr, EatToken());
+                    }
 
-                    case SyntaxKind.DotToken:
-                        expr = _syntaxFactory.MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression, expr, this.EatToken(), this.ParseSimpleName(NameOptions.InExpression));
+                    // Expand to consume the `dependent_access *` continuations.
+                    if (tryParseDependentAccess(expr) is ExpressionSyntax expandedExpression)
+                    {
+                        expr = expandedExpression;
                         continue;
+                    }
 
-                    case SyntaxKind.QuestionToken:
-                        return !CanStartConsequenceExpression()
-                            ? expr
-                            : _syntaxFactory.ConditionalAccessExpression(
-                                expr,
-                                operatorToken: this.EatToken(),
-                                ParseConsequenceSyntax());
+                    if (TryParseConditionalAccessExpression(expr, out var conditionalAccess))
+                        return conditionalAccess;
```

Comment:
I think this is correct as I expect the inner 'TryParseConditionalAccessExpression' call to consume any remaining dependent-access. But I may do some further refactor in my PR to at least assert this.